src/main/java/ecommerce/controllers/CurrentCartController.java

```diff
@@ -25,6 +25,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
         ShoppingCart currentShoppingCart = shoppingCartService.getByUserId(USER_ID);
         List<Product> products = currentShoppingCart.getProducts();
         req.setAttribute("products", products);
+        req.setAttribute("shoppingCartId", shoppingCartService.getByUserId(USER_ID).getId());
```

Comment:
you are calling database twice. Let's save shoppingCart to local variable

Reply:
done